wger/core/models.py

```diff
@@ -218,6 +221,31 @@ class UserProfile(models.Model):
     Default duration of workout pauses in the gym view
     """
 
+    @property
+    def is_trustworthy(self):
+        """
+        Returns true for trustworthy if they've had an account for more than three weeks.
+        More trustworthiness criteria could be added in future.
+        """
+
+        days_since_joined = datetime.date.today() - self.user.date_joined.date()
+        days_in_three_weeks = 21
+
+        if days_since_joined.days > days_in_three_weeks:
+            # perms will be updated only once, after they hit three week mark
+            image_perm = Permission.objects.get(codename='add_exerciseimage')
+            self.user.user_permissions.add(image_perm)
```

Comment:
I'd rather not give the user the add_exerciseimage permission just yet, we can simply do the dynamic checks

Reply:
Understandable! That was our original thought but we struggled to find where to do the dynamic check that would allow them to actually see the image adding form, so I guess do you have any guidance on where that would be? Originally we thought it was in wger/exercises/views/images.py class ExerciseImageAddView but couldn't see how to incorporate it without adding permissions.

Reply:
yeah, looking at the code it's a bit trickier that what I thought. One possibility would be to manually check for permissions or the trustworthy attribute by overwriting the dispatch method (you can then remove the PermissionRequiredMixin mixin and the permission_required). Something like this is done elsewhere, e.g. in the WgerPermissionMixin or WgerFormMixin where it checks the ownership of database objects.